library/src/main/kotlin/com/livefront/gsonkotlinadapter/util/KParameter.kt

```diff
@@ -29,6 +29,22 @@ internal val KParameter.defaultValue: Any?
         }
     }
 
+/**
+ * Returns the `true` if this [KParameter] is a primitive, `false` otherwise.
+ */
+internal val KParameter.isPrimitive: Boolean
+    get() = when (type.classifier as? KClass<*>) {
+        Boolean::class,
+        Byte::class,
+        Char::class,
+        Double::class,
+        Float::class,
+        Integer::class,
+        Long::class,
+        Short::class -> true
+        else -> false
+    }
```

Comment:
I see that https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.jvm/java-primitive-type.html exists, but it looks like it does some strange things with `java.lang.Void`, so I think this is fine.

Reply:
Something like this:
```kotlin
internal val KParameter.isPrimitive: Boolean
    get() = (type.classifier as? KClass<*>)?. javaPrimitiveType != null
```

🤔

Reply:
After looking at the code, I think I would just leave it.